### PR TITLE
Adjust ffmpeg version detection in admin panel

### PR DESCRIPTION
**Pull request by vmstan**
The static builds linked off of ffmpeg.org use a non-X.Y.Z format that the previous regex was failing on. This looks for number format or the next block of text instead of matching the number format. If neither function it'll return 0.0.0.

image

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
I'm not sure why we should ignore anything that isn't strictly point-separated numerics, `n7.0.1` is still a ffmpeg version number as far as `ffmpeg` is concerned.

https://superuser.com/a/1609819 gives an interesting reply regarding how to cleanly parse

There are also things that I'm not sure how I missed when reviewing #30710:
- I believe those are the only places where we are shelling out using the backticks; we normally use `terrapin`
- we allow specifying custom ffmpeg binaries through `FFMPEG_BINARY` and `FFPROBE_BINARY` (see e.g. `app/lib/video_metadata_extractor.rb` using `ffprobe_binary`) and this code ignores that

File: app/lib/admin/metrics/dimension/software_versions_dimension.rb [hunk 1]
```diff
@@ -98,7 +98,10 @@ def imagemagick_version
   end
 
   def ffmpeg_version
-    version = `ffmpeg -version`.match(/ffmpeg version ([\d\.]+)/)[1]
+    version_output = Terrapin::CommandLine.new(Rails.configuration.x.ffmpeg_binary, '-version').run
+    version_match = version_output.match(/ffmpeg version ([\d\.]+|[^\s]+)/)
+
+    version = version_match ? version_match[1] : '0.0.0'
```
Comment (ClearlyClaire) [inline app/lib/admin/metrics/dimension/software_versions_dimension.rb:104]: The following should be future-proof and avoid relying on a specific version format:
```suggestion
    version_output = Terrapin::CommandLine.new(Rails.configuration.x.ffprobe_binary, '-show_program_version -v 0 -of json').run
    version = Oj.load(version_output, mode: :strict, symbol_keys: true).dig(:program_version, :version)
```

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Thanks for the improvements!

I think you also need to update the `rescue` statement to something like `rescue Terrapin::CommandNotFoundError, Terrapin::ExitStatusError`.

And I again think nightly and such versions should be reported instead of displaying `0.0.0`. The code suggestion should do it. If you accept it, you should also add `Oj:ParseError`to the rescue line.

File: app/lib/admin/metrics/dimension/software_versions_dimension.rb (hunk 1 above)
Comment (vmstan) [reply app/lib/admin/metrics/dimension/software_versions_dimension.rb:104]: I didn't realize the ffprobe binary was an env option as well, but makes more sense to do it this way.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
The `rescue` statement a few lines below still needs to be changed to `rescue Terrapin::CommandNotFoundError, Terrapin::ExitStatusError, Oj::ParseError`

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**